packages/gatsby-cli/src/reporter/loggers/ipc/index.ts

```diff
@@ -0,0 +1,57 @@
+import { onLogAction } from "../../redux/index"
+import { ISetStatus, ActionsUnion } from "../../redux/types"
+import { Actions, LogLevels } from "../../constants"
+import stripAnsi from "strip-ansi"
+
+const isISetStatus = (action: ActionsUnion): action is ISetStatus =>
+  typeof action.payload === `string`
+
+/**
+ * Payload can either be a String or an Object
+ * See more at integration-tests/structured-logging/__tests__/to-do.js
+ */
+const sanitizeAction = (action: ActionsUnion): ActionsUnion => {
+  if (isISetStatus(action)) {
+    return action
+  }
+
+  if (`text` in action.payload && action.payload.text) {
+    action.payload.text = stripAnsi(action.payload.text)
+  }
+  if (`statusText` in action.payload && action.payload.statusText) {
+    action.payload.statusText = stripAnsi(action.payload.statusText)
+  }
+
+  return action
```

Comment:
I think there is a functional change going on here. In the previous version, we were creating a new object and spreading values into it. This had the benefit that we weren't mutating the underlying object.

I think there is a risk here that this function in it's new state would actually be modifying values that are in redux state (thanks JS object referencing.)

Let's change this to create a new object instead of modifying the one passed in.